compiler/src/iree/compiler/Codegen/Dialect/GPU/TargetUtils/ConfigUtils.cpp

```diff
@@ -123,6 +123,56 @@ LogicalResult setDataTiledMultiMmaLoweringConfig(
       workgroupSize, targetSubgroupSize, pipelineConfig);
 }
 
+/// Sort the MMA intrinsics by following precedence rules:
+///   1) k-alignment. We prefer intrinsics that can evenly divide the K
+///   dimension of the problem.
+///   2) M/N-alignment. We prefer intrinsics that can evenly divide the M and N
+///   dimensions of the problem.
+///   3) Intrinsic with larger gemm size.
+///   4) Intrinsic with larger K size.
+static void sortMMAIntrinsics(GPUMatmulShapeType problem,
+                              SmallVector<GPUIntrinsicType> &intrinsics) {
+  llvm::sort(intrinsics, [&](const GPUMatmulShapeType &lhs,
+                             const GPUMatmulShapeType &rhs) {
+    // Prefer K-aligned intrinsics.
+    int64_t lhsKAligned =
+        problem.kSizes.back() % lhs.kSizes.back() == 0 ? 1 : 0;
+    int64_t rhsKAligned =
+        problem.kSizes.back() % rhs.kSizes.back() == 0 ? 1 : 0;
+    if (lhsKAligned != rhsKAligned) {
+      return lhsKAligned > rhsKAligned;
+    }
+
+    // If K alignment is the same, prefer the intrinsic that aligns M and N.
+    int64_t lhsMNAligned = (problem.mSizes.back() % lhs.mSizes.back() == 0 &&
+                            problem.nSizes.back() % lhs.nSizes.back() == 0)
+                               ? 1
+                               : 0;
+    int64_t rhsMNAligned = (problem.mSizes.back() % rhs.mSizes.back() == 0 &&
+                            problem.nSizes.back() % rhs.nSizes.back() == 0)
+                               ? 1
+                               : 0;
+    if (lhsMNAligned != rhsMNAligned) {
+      return lhsMNAligned > rhsMNAligned;
+    }
+
+    auto intrinsicArea = [&](const GPUMatmulShapeType &intrinsic) {
+      return (ShapedType::getNumElements(intrinsic.mSizes) +
+              ShapedType::getNumElements(intrinsic.nSizes)) *
+             ShapedType::getNumElements(intrinsic.kSizes);
+    };
+    auto lhsArea = intrinsicArea(lhs);
```

Comment:
Nit: name the integer types

Reply:
How about lhsGemmSize? I'm not good at naming stuff, so let me know if you have better suggestions!